backend/src/api/model/series.rs

```diff
@@ -327,6 +349,193 @@ impl Series {
+            ", &[&series.key, &title, &description]).await?;
+
+            Self::load_by_id(id, context)
+                .await?
+                .ok_or_else(|| err::invalid_input!(
+                    key = "series.metadata.not-found",
+                    "series not found",
+                ))
+        } else {
+            warn!(
+                series_id = %id,
+                "Failed to update series metadata, OC returned status: {}",
+                response.status(),
+            );
+            Err(err::opencast_error!("Opencast API error: {}", response.status()))
+        }
+    }
+
+    pub(crate) async fn delete(id: Id, context: &Context) -> ApiResult<Series> {
+        let series = Self::load_for_api(
+            id,
+            context,
+            err::invalid_input!(
+                key = "series.delete.not-found",
+                "series not found",
+            ),
+            err::not_authorized!(
+                key = "series.delete.not-allowed",
+                "deletion not allowed",
+            )
+        ).await?;
+
+        info!(series_id = %id, "Attempting to send request to delete series in Opencast");
+
+        context.db.execute("\
+            update all_series \
+            set tobira_deletion_timestamp = current_timestamp \
+            where id = $1 \
+        ", &[&series.key]).await?;
+        // Todo: Consider optimistically updating events as well.
+
+
+        let oc_client = context.oc_client.clone();
+        let series_clone = series.clone();
+
+        // Unfortunately we can't wait for the http response. The Opencast delete endpoint will
+        // automatically start `republish metadata` workflows for all events in the series, which
+        // takes roughly 10 seconds per event, and only returns once all have finished.
+        // This would block the request for a long time.
+        tokio::spawn(async move {
+            let response = match oc_client.delete(&series_clone).await {
+                Ok(response) => response,
+                Err(e) => {
+                    error!("Failed to send delete request: {}", e);
+                    return;
+                }
+            };
+
+            if response.status() == StatusCode::NO_CONTENT {
+                info!(series_id = %id, "Deletion successfully requested");
+            } else {
+                // This is kinda pointless. Depending on the number of videos, the request takes
+                // super long to respond and will potentially return with `504 Gateway Timeout`.
+                // This is not necessarily an error in this case as the deletion could still be
+                // in progress.
+                warn!(
+                    series_id = %id,
+                    "Failed to delete series, Opencast returned status: {}",
+                    response.status()
+                );
+            }
+            // Todo: Consider reverting DB changes on error or unexpected response.
```

Comment:
Leave the todo here, but just my opinion: with a new data model, all this madness will disappear. We already said that Tobira will do a quick'n'dirty workaround around Opencasts crazy behavior. Sync should always fix everything eventually.